middleware/http/opa/middleware_test.go

```diff
@@ -232,3 +258,55 @@ func TestOpaPolicy(t *testing.T) {
 		})
 	}
 }
+
+func TestStatus_UnmarshalJSON(t *testing.T) {
+	type testObj struct {
+		Value Status `json:"value,omitempty"`
+	}
+	tests := map[string]struct {
+		jsonBytes   []byte
+		expectValue Status
+		expectError bool
+	}{
+		"int value": {
+			jsonBytes:   []byte(`{"value":2}`),
+			expectValue: Status(2),
+			expectError: false,
+		},
+		"string value": {
```

Comment:
Per comment in the code, I'd also add a test case to check that we reject non-integer values
```go
		"invalid float value": {
			jsonBytes:   []byte(`{"value":2.9}`),
			expectError: true,
		},
```

Reply:
Done.